shpc/main/modules/module.py

```diff
@@ -9,17 +9,32 @@
 
 
 class Module:
-    def __init__(self, name):
+    def __init__(self, config):
         """
         New module metadata and shared functions.
 
         This should be created by base.py new_module to ensure the same
         container and settings are carried forward here.
         """
-        self.name = name
```

Comment:
If config can be a config _or_ module name, let's keep the variable named to `name`. And could we visually break up the logic with a function? E.g.,

```python
class Module:
    def __init__(self, name):
        # Note for muffato: this can be set to a default of None, and overridden given it's a config

        # Move the defaults (currently below) up to the top, and they could be over-ridden by set_config
        self._uri = None

        # We can technically do this for a default, and it will be updated if we've been given a config instead!
        self.name = name

        if not isinstance(self.name, str):
            self.load_config(name)
        self.module_basepath = self.name.replace(':', os.sep)
```
It's functionally the same, but maybe a little more modular / easier to read.